internal/server/repository_worker.go

```diff
@@ -558,3 +558,156 @@ func (r *Repository) CreateWorker(ctx context.Context, worker *Worker, opt ...Op
 	}
 	return returnedWorker, nil
 }
+
+// AddWorkerTags adds specified api tags to the repo worker and returns its new tags.
+// No options are currently supported.
+func (r *Repository) AddWorkerTags(ctx context.Context, workerId string, workerVersion uint32, tags []*Tag, _ ...Option) ([]*Tag, error) {
+	const op = "server.(Repository).AddWorkerTags"
+	switch {
+	case workerId == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "worker public id is empty")
+	case workerVersion == 0:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing version")
+	case len(tags) == 0:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "no tags provided")
+	}
+
+	worker, err := lookupWorker(ctx, r.reader, workerId)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	if worker == nil {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("no worker found with public id %s", workerId))
+	}
+
+	newTags := worker.apiTags
```

Comment:
I'd prefer not renaming them to tags because that's the name of the input parameter. It made sense to me because they are the worker's new set of tags after the add tags operation. I'll keep it as is because I don't have a better name in mind right now, but I am open to suggestions